bcbio/ngsalign/postalign.py

```diff
@@ -198,14 +200,16 @@ def correct_umis(data):
     # Improve speeds by avoiding compression read/write bottlenecks
     io_opts = "--async-io=true --compression=0"
     umis_whitelist = tz.get_in(["config", "algorithm", "correct_umis"], data)
-    umi_method, umi_tag = _check_umi_type(input_bam)
-
-    cmd = ("unset JAVA_HOME && "
-           "fgbio {jvm_opts} {io_opts} CorrectUmis "
-           "-t {umi_tag} -m 3 -d 1 -x "
-           "-U {umis_whitelist} "
-           "-i {input_bam} -o {output_bam}")
-    do.run(cmd.format(**locals()), "Correcting UMIs")
+
+    if not utils.file_exists(output_bam):
+        umi_method, umi_tag = _check_umi_type(input_bam)
+        cmd = ("unset JAVA_HOME && "
+               "fgbio {jvm_opts} {io_opts} CorrectUmis "
+               "-t {umi_tag} -m 3 -d 1 -x "
+               "-U {umis_whitelist} "
+               "-i {input_bam} -o /dev/stdout | samtools view -bh > {output_bam}")
```

Comment:
Thanks Sergey! It would be good to do a `config_utils.get_program('fgbio", data)` and `config_utils.get_program("samtools", data)` here to make sure we are picking up the bcbio-installed versions. See https://github.com/bcbio/bcbio-nextgen/blob/bd4a01005f8041cb5c876f287eeb30bb3fb0ae9c/bcbio/chipseq/__init__.py#L51-L56 for an example use.

Reply:
There are parts in the code where we don't do this, and we've been slowly trying to fix them all, so if you happen to see them when you are doing other stuff, fixing it would be great.